go/vt/vtgate/evalengine/eval_result.go

```diff
@@ -31,12 +33,15 @@ import (
 
 type (
 	EvalResult struct {
-		typ       querypb.Type
-		collation collations.TypedCollation
-		numval    uint64
-		bytes     []byte
-		tuple     *[]EvalResult
-		decimal   *decimalResult
+		expr Expr
+		env  *ExpressionEnv
+
+		typ3       querypb.Type
```

Comment:
What does the `3` in these fields mean?

Reply:
I forgot to clean this up. I've given the fields more clean names and commented them.